From 886bbea62957c948d43ff4e1a1da3d8ac5a1139a Mon Sep 17 00:00:00 2001
From: Luo Meng <luomeng12@huawei.com>
Date: Tue, 30 Aug 2022 18:30:30 +0800
Subject: [PATCH] Revert "NFS: Use of mapping_set_error() results in spurious
 errors"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 886bbea62957c948d43ff4e1a1da3d8ac5a1139a
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5KK4D
CVE: NA

--------------------------------

This reverts commit f4f117026d586db38c025cd95b6094ff8de326df.

7f3945b4aab5 (NFS: Use of mapping_set_error() results in spurious errors)
delete mappint->flags because of 67dd23f9e6fb (nfs: ensure correct writeback
errors are returned on close()). However 67dd23f9e6fb has been reverted, use
mappint->flags to return error.

Signed-off-by: Luo Meng <luomeng12@huawei.com>
Reviewed-by: Zhang Xiaoxu <zhangxiaoxu5@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/nfs/write.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/fs/nfs/write.c b/fs/nfs/write.c
index 5d07799513a6..e8f6ce88b1fb 100644
--- a/fs/nfs/write.c
+++ b/fs/nfs/write.c
@@ -310,10 +310,7 @@ static void nfs_mapping_set_error(struct page *page, int error)
 	struct address_space *mapping = page_file_mapping(page);
 
 	SetPageError(page);
-	filemap_set_wb_err(mapping, error);
-	if (mapping->host)
-		errseq_set(&mapping->host->i_sb->s_wb_err,
-			   error == -ENOSPC ? -ENOSPC : -EIO);
+	mapping_set_error(mapping, error);
 	nfs_set_pageerror(mapping);
 }
 
-- 
2.34.1

